src/graph/graph.user.ts

```diff
@@ -107,3 +109,58 @@ export async function getUsersForUserIds(graph: IGraph, userIds: string[]): Prom
     }
   }
 }
+
+/**
+ * Returns a Promise of Graph Users array associated with the people queries array
+ *
+ * @export
+ * @param {IGraph} graph
+ * @param {string[]} peopleQueries, an array of string ids
+ * @returns {Promise<User[]>}
+ */
+export async function getUsersForPeopleQueries(graph: IGraph, peopleQueries: string[]): Promise<User[]> {
+  if (!peopleQueries || peopleQueries.length === 0) {
+    return [];
+  }
+
+  const batch = graph.createBatch();
+  const people = [];
+
+  for (const personQuery of peopleQueries) {
+    if (personQuery !== '') {
+      batch.get(personQuery, `/me/people?$search="${personQuery}"`, ['people.read']);
+    }
+  }
+
+  try {
+    const response = await batch.execute();
+
+    for (const personQuery of peopleQueries) {
+      const person = response[personQuery];
+      if (person) {
+        people.push(person.value[0]);
+      }
+    }
+
+    return people;
+  } catch (_) {
+    try {
+      for (const personQuery of peopleQueries) {
+        if (personQuery !== '') {
+          const person = (await findPerson(graph, personQuery)) as IDynamicPerson[];
+          if (person && person.length) {
+            people.push(person[0]);
+
+            const image = await getPersonImage(graph, person[0]);
+            if (image) {
+              person[0].personImage = image;
+            }
+          }
+        }
+      }
```

Comment:
Optimize this section to run the promises in paralel and remove fetching images (since we are not fetching images above):

```suggestion
      const promises = peopleQueries.map(personQuery =>
        personQuery !== '' ? findPerson(graph, personQuery) : Promise.resolve(null)
      );
      const results = await Promise.all(promises);

      for (const person of results) {
        if (person && person.length) {
          people.push(person[0]);
        }
      }
```